packages/plugins/jwt/src/index.ts

```diff
@@ -140,9 +109,53 @@ export function useJwt(options: JwtPluginOptions): Plugin {
       const payload = payloadByRequest.get(context.request)
       if (payload != null) {
         extendContext({
-          [contextFieldName]: payload,
+          [extendContextField]: payload,
         })
       }
     },
   }
 }
+
+function unauthorizedError(message: string) {
+  return new GraphQLError(message, {
+    extensions: {
+      http: {
+        status: 401,
+      },
+    },
+  })
+}
+
+function verify(
+  token: string,
+  signingKey: string,
+  options: Parameters<typeof jsonwebtoken.verify>[2],
+) {
+  return new Promise((resolve, reject) => {
+    jsonwebtoken.verify(token, signingKey, options, (err, result) => {
+      if (err) {
+        // Should we expose the error message? Perhaps only in development mode?
+        reject(unauthorizedError('Failed to decode authentication token'))
```

Comment:
If the token is incorrect, this is the user's fault right? So it should be ok to expose it then, no?

Reply:
Yes, but in security context, you always want to feedback as less information as possible. This prevents attackers from testing things around to better understand your implemtation and potential flaws.
It's pretty much the same principle as Error Masking feature for me.

Reply:
Hmm good point. @n1ru4l @dotansimha what do you think?

Reply:
For me it is fine because the algorithm, public key and so on are verified and they are unique enough in my opinion.